server/gitlab/gitlab.go

```diff
@@ -47,23 +48,70 @@ type gitlab struct {
 	checkGroup        func(projectNameWithGroup string) error
 }
 
+// Scope identifies the scope of a webhook
+type Scope int
+
+const (
+	//Group is a type for group hooks
+	Group Scope = iota
+	//Project is a type for project hooks
+	Project
+)
+
+func (s Scope) String() string {
+	return [...]string{"group", "project"}[s]
+}
+
+// WebhookInfo Provides information about group or project hooks.
+type WebhookInfo struct {
+	ID                       int
+	URL                      string
+	ConfidentialNoteEvents   bool
+	PushEvents               bool
+	IssuesEvents             bool
+	ConfidentialIssuesEvents bool
+	MergeRequestsEvents      bool
+	TagPushEvents            bool
+	NoteEvents               bool
+	JobEvents                bool
+	PipelineEvents           bool
+	WikiPageEvents           bool
+	EnableSSLVerification    bool
+	CreatedAt                *time.Time
+	Scope                    Scope
+}
+
+//AddWebhookOptions is a paramater object with options for creating a project or group hook.
+type AddWebhookOptions struct {
+	URL                      string
+	ConfidentialNoteEvents   bool
+	PushEvents               bool
+	IssuesEvents             bool
+	ConfidentialIssuesEvents bool
+	MergeRequestsEvents      bool
+	TagPushEvents            bool
+	NoteEvents               bool
+	JobEvents                bool
+	PipelineEvents           bool
+	WikiPageEvents           bool
+	EnableSSLVerification    bool
+	Token                    string
+}
```

Comment:
This file begins go get clustered. Let's  move the webhook code to `server/gitlab/webhook.go`.

Reply:
Thanks for the nudge, I wanted to do that as well but was hesitant about having two webhook.go files in the same project.